pkg/traceql/ast.go

```diff
@@ -250,27 +250,29 @@ func newSpansetOperation(op Operator, lhs SpansetExpression, rhs SpansetExpressi
 func (SpansetOperation) __spansetExpression() {}
 
 type SpansetFilter struct {
-	Expression FieldExpression
+	Expression          FieldExpression
+	matchingSpansBuffer []Span
 }
 
-func newSpansetFilter(e FieldExpression) SpansetFilter {
-	return SpansetFilter{
+func newSpansetFilter(e FieldExpression) *SpansetFilter {
+	return &SpansetFilter{
 		Expression: e,
 	}
 }
 
 // nolint: revive
-func (SpansetFilter) __spansetExpression() {}
+func (_ *SpansetFilter) __spansetExpression() {}
 
-func (f SpansetFilter) evaluate(input []*Spanset) ([]*Spanset, error) {
+func (f *SpansetFilter) evaluate(input []*Spanset) ([]*Spanset, error) {
 	var output []*Spanset
```

Comment:
similar buffer shenanigans for output?

Reply:
I think that's not possible because of the spanset pooling added in vparquet2, but will try it.

Reply:
I don't know what I was thinking. We repool the individual spansets not the slice so added and it works just fine.